bottlecap/src/lifecycle/invocation/triggers/mod.rs

```diff
@@ -0,0 +1,51 @@
+use std::{collections::HashMap, hash::BuildHasher};
+
+use datadog_trace_protobuf::pb::Span;
+use serde::{ser::SerializeMap, Serializer};
+use serde_json::Value;
+
+pub mod api_gateway_http_event;
+
+pub trait Trigger: Sized {
+    fn new(payload: Value) -> Option<Self>;
+    fn is_match(payload: &Value) -> bool;
```

Comment:
Would it be more convenient to have a general matching function returning a type instead of is_match for every subclass? To avoid doing is_match till something matches and maybe hiding bugs when a match hides running other match logic

Reply:
I thought so, but I wanted to not have to separate payload-specific logic into its own trait, I'm still exploring the idea, but I think this might change depending on how it would look like when we add more cases.

How would you change this code? My main constraints are: not deserialize on every match, just deserialize once into `serde_json::Value` and then try to find the actual type of the trigger, wdyt?